Objects/exceptions.c

```diff
@@ -2980,15 +2992,22 @@ UnicodeEncodeError_init(PyObject *self, PyObject *args, PyObject *kwds)
     if (!PyArg_ParseTuple(args, "UUnnU",
                           &err->encoding, &err->object,
                           &err->start, &err->end, &err->reason)) {
```

Comment:
I suggest to store values in local variables, check start, and only then modify *err*. Something like:

```c
err->encoding = Py_NewRef(encoding);
...
err->start = start;
...
```

It would leave err->encoding, err->object and err->reason unchanged if start is invalid, and avoid the need for an error label.

Same remark for UnicodeDecodeError_init() and UnicodeTranslateError_init().

Reply:
I'm sometimes having a bit of trouble with `Py_NewRef` so thank you for your suggestion! (it's simpler to know when to decref than to know when to incref IMO).